airflow/operators/sftp_to_gcs.py

```diff
@@ -0,0 +1,219 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains SFTP to Google Cloud Storage operator.
+"""
+import os
+from tempfile import NamedTemporaryFile
+from typing import Optional, Union
+
+from airflow import AirflowException
+from airflow.contrib.hooks.sftp_hook import SFTPHook
+from airflow.gcp.hooks.gcs import GoogleCloudStorageHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+WILDCARD = "*"
+
+
+class SFTPToGoogleCloudStorageOperator(BaseOperator):
+    """
+    Transfer files to Google Cloud Storage from SFTP server.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:SFTPToGoogleCloudStorageOperator`
+
+    :param source_path: The sftp remote path. This is the specified file path
+        for downloading the single file or multiple files from the SFTP server.
+        You can use only one wildcard within your path. The wildcard can appear
+        inside the path or at the end of the path.
+    :type source_path: str
+    :param destination_bucket: The bucket to upload to.
+    :type destination_bucket: str
+    :param destination_path: The destination name of the object in the
+        destination Google cloud storage bucket.
+        If destination_path is not provided file/files will be placed in the
+        main bucket path.
+        If a wildcard is supplied in the destination_path argument, this is the
+        prefix that will be prepended to the final destination objects' paths.
+    :type destination_path: str
+    :param gcp_conn_id: (Optional) The connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    :param sftp_conn_id: The sftp connection id. The name or identifier for
+        establishing a connection to the SFTP server.
+    :type sftp_conn_id: str
+    :param delegate_to: The account to impersonate, if any
+    :type delegate_to: str
+    :param mime_type: The mime-type string
+    :type mime_type: str
+    :param gzip: Allows for file to be compressed and uploaded as gzip
+    :type gzip: bool
+    :param move_object: When move object is True, the object is moved instead
+        of copied to the new location. This is the equivalent of a mv command
+        as opposed to a cp command.
+    :type move_object: bool
+
+       :Example:
+
+    The following Operator would copy a single file named
```

Comment:
Because we have a guide, it is better not to include explanations here, because the guide is of better quality.  The guides are system tested.

Reply:
Sure. I will delete it.

Reply:
Sure. I will delete it.